Space Pirates Enemies

KyleLotilla opened the pull request:
Added a new type of enemies: the SpacePirates. There's only two types of these enemies currently: The Peashooter and the Rocket Launcher. I will add another type for the shotgun, once the shotgun is finished. Also, the current movement is it first moves into a defined bounds and then it moves randomly inside these bounds. As shown here:
![enemy](https://user-images.githubusercontent.com/44867791/131211673-71734f47-86e2-4afb-a2ce-39704f63aa76.gif)

KyleLotilla committed:
Added Peashooter Enemy Ship and Rocket Launcher Enemy Ship. Changed Projectile Movement to not consider rotations anymore.

Cytoplazam commented:
Projectiles were still spawning even though the enemies are dead. Add fourth frame of the enemy ship animations (Similar to the player ship animation)

KyleLotilla committed:
Fixed animations of Enemy Ships. Fixed Enemies not counting towards the EncounterSpawner. Fixed Enemy Ship shooting when exploded.

KyleLotilla commented:
I fixed the bugs listed above and I also fixed another bug with the ships not counting their deaths to the `EncounterSpawner`.

I'll go ahead and merge.
